tests/base-notebook/test_healthcheck.py

```diff
@@ -13,22 +13,48 @@
 
 
 @pytest.mark.parametrize(
-    "env",
+    "env,cmd,workdir",
     [
-        None,
-        ["DOCKER_STACKS_JUPYTER_CMD=lab"],
-        ["DOCKER_STACKS_JUPYTER_CMD=notebook"],
-        ["DOCKER_STACKS_JUPYTER_CMD=server"],
-        ["DOCKER_STACKS_JUPYTER_CMD=nbclassic"],
-        ["RESTARTABLE=yes"],
-        ["JUPYTER_PORT=8171"],
-        ["JUPYTER_PORT=8117", "DOCKER_STACKS_JUPYTER_CMD=notebook"],
+        (None, None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=lab"], None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=notebook"], None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=server"], None, None),
+        (["DOCKER_STACKS_JUPYTER_CMD=nbclassic"], None, None),
+        (["RESTARTABLE=yes"], None, None),
+        (["JUPYTER_PORT=8171"], None, None),
+        (["JUPYTER_PORT=8117", "DOCKER_STACKS_JUPYTER_CMD=notebook"], None, None),
+        (None, ["start-notebook.sh", "--NotebookApp.base_url=/test"], None),
+        (None, ["start-notebook.sh", "--NotebookApp.base_url=/test/"], None),
+        (["GEN_CERT=1"], ["start-notebook.sh", "--NotebookApp.base_url=/test"], None),
+        (
+            ["GEN_CERT=1", "JUPYTER_PORT=7891"],
+            ["start-notebook.sh", "--NotebookApp.base_url=/test"],
+            None,
+        ),
+        (["NB_USER=testuser", "CHOWN_HOME=1"], None, "/home/testuser"),
+        (
+            ["NB_USER=testuser", "CHOWN_HOME=1"],
+            ["start-notebook.sh", "--NotebookApp.base_url=/test"],
+            "/home/testuser",
+        ),
+        (
+            ["NB_USER=testuser", "CHOWN_HOME=1", "JUPYTER_PORT=8123"],
+            ["start-notebook.sh", "--NotebookApp.base_url=/test"],
+            "/home/testuser",
+        ),
     ],
 )
-def test_health(container: TrackedContainer, env: Optional[list[str]]) -> None:
+def test_health(
```

Comment:
Wow, I like the number of tests you have 👍 

I have two questions though:
1. Why do you need to change workdir?
2. We do `time.sleep(15)`. I think when we have this many tests it's better to return early (for example have a loop for 15 seconds maximum, which checks each 0.1 second, whether the status is healthy).

Reply:
1. I included it just so it resembles the example on https://jupyter-docker-stacks.readthedocs.io/en/latest/using/common.html#user-related-configurations more closely. It's not really needed actually, I tested without changing the workdir as well and it doesn't cause any problems. Should we not change it to simplify the test?
2. That's a good point. I can try to implement it. When testing locally it didn't feel like it took that long though, I guess pytest runs those tests in parallel

Reply:
1. I think if it doesn't change behaviour in any visible way, it's better to remove it.